app/views/films/index.html.erb

```diff
@@ -17,163 +17,257 @@
+<!--            </div>-->
 
-        <%# end %>
-
-<!--              </div>-->
           <%# end %>
+<!--        </div>-->
 
-      <% if !flash[:notice].blank? %>
-        <%= flash[:notice] %>
-      <% end %>
-      <br>
-      <%= link_to "Add Movie",  films_new_path  %>
-      <%#= button_to "Add Movie", films_new_path, class: "btn btn-primary" %>
-
-      <% @films.in_groups_of(5, false) do |row| %>
-        <div class="movies_box">
-
-          <% row.compact.each do |film| %>
-            <div class="movies_card">
+      <%# end %>
 
-              <div class="movies_image">
-                <div class="hover">
-                  <i class="fa-solid fa-star"></i>
-                  <p class="rate"><%= film.rating %></p>
-                  <p class="categories"><%= film.genre %></p>
-                  <%=link_to "View Details", film_path(film), :class => 'btn btn-primary'%>
-                </div>
 
-<!--                <img src="<%#= film.images.picture.first.url %>">-->
+<!--    </section>-->
 
-                <img src="<%= film.images.first.picture.url %>">
+   <hr>
 
+<caption><h1>Movies From Api</h1></caption>
+<table class="table table-bordered">
+  <thead>
+  <tr>
+    <th scope="col">#</th>
+    <th scope="col">Title</th>
+    <th scope="col">Release Date</th>
+    <th scope="col">Vote Average</th>
+    <th scope="col">Genres</th>
+    <th scope="col">Poster</th>
+  </tr>
+  </thead>
+  <tbody>
+  <% @movie_detail.in_groups_of(5, false) do |row| %>
+    <% row.compact.each_with_index do |movie, index| %>
+      <tr>
+        <th scope="row"><%= index + 1 %></th>
+        <td><%= movie['title'] %></td>
+        <td><%= movie['release_date'] %></td>
+        <td><%= movie['vote_average'] %></td>
+        <td><%= movie['genres'] %></td>
+        <td style="max-width: 50px;"><%= image_tag movie['poster_path'], style: "max-width: 100%; height: auto;" %></td>
+      </tr>
+    <% end %>
+  <% end %>
+  </tbody>
+</table>
 
-                <%#= image_tag(film.picture) %>
-              </div>
 
-              <div class="text">
-                <p class="name"><%= film.title %></p>
-                <p class="year"><%= film.year %></p>
-              </div>
-            </div>
 
-          <% end %>
 
-        </div>
-      <% end %>
 
 
 
 
 
 
-<!--<h1>All Movies</h1>-->
-<%# @films.each do |row|%>
-<%# @films.in_groups_of(5, false) do |row| %>
-<!--  <div class="row">-->
-    <%# row.compact.each do |film| %>
 
-    <%#= image_tag(film&.movie, style:'width:19%') %>
-<!--    <h3><%#=film.title%></h3>-->
-<!--    <p><%#=film.year%></p>-->
-<!--    <p><%#=film.genre%></p>-->
-<!--    <p><%#=film.rating%></p>-->
 
-    <%# end %>
-<!--  </div>-->
-<!--        <br>-->
-<%# end %>
+<!--    <section>-->
+<!--      <h1>New movies From Apis, DB movies are not included</h1>-->
+      <%# @new_movies.in_groups_of(5, false) do |row| %>
+<!--        <div class="movies_box">-->
 
+          <%# row.compact.each do |movie| %>
```

Comment:
what is the purpose of using `.compact` here ?
explain why you feel need to use it here

Reply:
.compact method is use to remove any nil values from an array. 
ye tab use kia tha jab yts movies ko print krwana tha 5 movies in a row.
agr isko remove kr bhi dain gy to tab bhi chal jae ga ye.